src/OneExplorer.ts

```diff
@@ -0,0 +1,183 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+enum NodeType{
+  directory,
+  model,
+  config
+}
+
+interface Node {
+  type: NodeType, name: string, childs: Node[], uri: vscode.Uri
+}
+
+export class OneNode extends vscode.TreeItem {
+  constructor(
+      public readonly label: string,
+      public readonly collapsibleState: vscode.TreeItemCollapsibleState,
+      public readonly node: Node,
+  ) {
+    super(label, collapsibleState);
+
+    this.tooltip = `${this.label}`;
+
+    if (node.type === NodeType.config) {
+      this.iconPath = new vscode.ThemeIcon('gear');
+    }
+    else if (node.type === NodeType.directory){
+      this.iconPath = vscode.ThemeIcon.Folder;
+    }
+    else if (node.type === NodeType.model) {
+      this.iconPath = vscode.ThemeIcon.File;
+    }
+  }
+}
+
+export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
+  private _onDidChangeTreeData: vscode.EventEmitter<OneNode|undefined|void> =
+      new vscode.EventEmitter<OneNode|undefined|void>();
+  readonly onDidChangeTreeData: vscode.Event<OneNode|undefined|void> =
+      this._onDidChangeTreeData.event;
+
+  private cfgMap: Node|undefined;
+
+  constructor(private workspaceRoot: vscode.Uri|undefined) {
+    if (workspaceRoot !== undefined) {
+      this.cfgMap = this.getConfigMap(workspaceRoot);
+    }
+  }
+
+  // TODO(dayo): enable refresh command
+  refresh(): void {
+    this._onDidChangeTreeData.fire();
+  }
+
+  getTreeItem(element: OneNode): vscode.TreeItem {
+    element.command = { command: 'oneExplorer.openFile', title: "Open File", arguments: [element.node] };
+    return element;
+  }
+
+  getChildren(element?: OneNode): OneNode[]|Thenable<OneNode[]> {
+    if (this.cfgMap === undefined) {
+      vscode.window.showInformationMessage('No ONE model or config in empty workspace');
+      return Promise.resolve([]);
+    }
+
+    if (element) {
+      return Promise.resolve(this.getNode(element.node));
+    } else {
+      return Promise.resolve(this.getNode(this.cfgMap));
+    }
+  }
+
+  private getNode(node: Node): OneNode[] {
+    const toOneNode = (node: Node): OneNode => {
+      if (node.childs.length > 0) {
+        return new OneNode(node.name, vscode.TreeItemCollapsibleState.Collapsed, node);
+      } else {
+        return new OneNode(node.name, vscode.TreeItemCollapsibleState.None, node);
+      }
+    };
+
+    return node.childs.map(node => toOneNode(node));
+  }
+
+  private getConfigMap(rootPath: vscode.Uri): Node {
+    const node = {type: NodeType.directory, name: path.parse(rootPath.fsPath).base, dir: true, childs: [], uri: rootPath};
+    this.searchConfigs(node, path.dirname(rootPath.fsPath));
+    return node;
+  }
+
+  private searchConfigs(node: Node, dirPath: string) {
```

Comment:
similar with 2 in https://github.com/Samsung/ONE-vscode/pull/505/files#r858276639.
How about `searchNode`?